mimalloc/SCsub

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+from misc.utility.scons_hints import *
+
+Import("env")
+
+# For platforms not supported out-of-the-box by mimalloc, the following options
+# can be used to add support for them:
+# mimalloc_custom_prims (bool): If true, the prim.c file is not included. Important because otherwise it falls back to UNIX.
+# mimalloc_custom_tls_slot (bool): If true, mimalloc is built assuming primitives for TLS operations will be provided separately.
+# mimalloc_custom_get_thread_id (string): If non-empty, mimalloc will use the function named here to get the current thread id.
+
+
+mimalloc_sources = []
+
+if env["platform"] == "windows":
+    override_type = "windows"
+elif env.get("mimalloc_custom_prims", False):
+    override_type = "none"
+else:
+    override_type = "static"
+
+use_mingw_custom_tls_slot = override_type == "windows" and not env.msvc
+
+# Thirdparty source files
+
+env_thirdparty = env.Clone()
+env_thirdparty.disable_warnings()
+
+thirdparty_obj = []
+
+thirdparty_dir = "#thirdparty/mimalloc/"
+if override_type == "static":
+    thirdparty_sources = ["src/static.c"]
+    env_thirdparty.Append(CPPDEFINES=["MI_MALLOC_OVERRIDE"])
+else:
+    thirdparty_sources = [
+        "src/alloc.c",
+        "src/alloc-aligned.c",
+        "src/alloc-posix.c",
+        "src/arena.c",
+        "src/bitmap.c",
+        "src/heap.c",
+        "src/init.c",
+        "src/libc.c",
+        "src/options.c",
+        "src/os.c",
+        "src/page.c",
+        "src/random.c",
+        "src/segment-map.c",
+        "src/segment.c",
+        "src/stats.c",
+    ]
+    if not env.get("mimalloc_custom_prims", False):
```

Comment:
Will return `None` if not found and no default provided; works for falsy conditional
```suggestion
    if not env.get("mimalloc_custom_prims"):
```